examples/langchain/Gemini_LangChain_QA_Couchbase_WebLoad.ipynb

```diff
@@ -0,0 +1,938 @@
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": 13,
+      "metadata": {},
+      "outputs": [
+        {
+          "name": "stderr",
+          "output_type": "stream",
+          "text": [
+            "2025-04-23 11:18:50,232 - INFO - Creating new index 'vector-search-testing.shared.vector_search_gemini'...\n",
+            "2025-04-23 11:18:50,438 - INFO - Index 'vector-search-testing.shared.vector_search_gemini' successfully created/updated.\n"
+          ]
+        }
+      ],
+      "source": [
+        "try:\n",
+        "    scope_index_manager = cluster.bucket(CB_BUCKET_NAME).scope(SCOPE_NAME).search_indexes()\n",
+        "\n",
+        "    # Check if index already exists\n",
+        "    existing_indexes = scope_index_manager.get_all_indexes()\n",
+        "    index_name = index_definition[\"name\"]\n",
+        "\n",
+        "    if index_name in [index.name for index in existing_indexes]:\n",
+        "        logging.info(f\"Index '{index_name}' found\")\n",
+        "    else:\n",
+        "        logging.info(f\"Creating new index '{index_name}'...\")\n",
+        "\n",
+        "    # Create SearchIndex object from JSON definition\n",
+        "    search_index = SearchIndex.from_json(index_definition)\n",
+        "\n",
+        "    # Upsert the index (create if not exists, update if exists)\n",
+        "    scope_index_manager.upsert_index(search_index)\n",
+        "    logging.info(f\"Index '{index_name}' successfully created/updated.\")\n",
+        "\n",
+        "except QueryIndexAlreadyExistsException:\n",
+        "    logging.info(f\"Index '{index_name}' already exists. Skipping creation/update.\")\n",
+        "except ServiceUnavailableException:\n",
+        "    raise RuntimeError(\"Search service is not available. Please ensure the Search service is enabled in your Couchbase cluster.\")\n",
+        "except InternalServerFailureException as e:\n",
+        "    logging.error(f\"Internal server error: {str(e)}\")\n",
+        "    raise"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {},
+      "source": [
+        "#  Setting Up the Couchbase Vector Store\n",
+        "A vector store is where we'll keep our embeddings. Unlike the FTS index, which is used for text-based search, the vector store is specifically designed to handle embeddings and perform similarity searches. When a user inputs a query, the search engine converts the query into an embedding and compares it against the embeddings stored in the vector store. This allows the engine to find documents that are semantically similar to the query, even if they don't contain the exact same words. By setting up the vector store in Couchbase, we create a powerful tool that enables our search engine to understand and retrieve information based on the meaning and context of the query, rather than just the specific words used."
```

Comment:
Something seems off with the comparison to FTS index cause technically, vector index is part of the same service. Might be worth getting rid of the comparison.